### PR TITLE
[READY] issue-128 - Adding golden files for template validation on openwrt

### PR DESCRIPTION
**Pull request by sarcasticadmin**
## Description of PR
@owendelong @kylerisse @davidelang 

Related to: #128 

Golden files will now be used to validate the output of `gomplate` for the openwrt configs. Some of the work that makes this possible was just merged into master for `gomplate`. Mainly: https://github.com/hairyhenderson/gomplate/pull/468 So Ill be waiting eagerly for a new version to drop. For now it requires building the `gomplate` binary directory with `go get`

Specific commit that enabled this functionality was: d1873444c90e68b5e8ba7492fb83abd36d7ab0ea

## Previous Behavior
No automated testing for generated files 

## New Behavior
* Automated testing of generated openwrt config files
* Renamed `openwrt.yaml.example` to `openwrt-example.yaml` (see commit message)
* Added `.gomplateignore` file to make sure it doesnt incldue `.swp` if you have a template open in another window
* Updated the documentation

## Tests
How was this PR tested?

```
cd tests/unit/openwrt
sh test.sh
```
If it returns 0 then there's not a diff from the `./tmp` dir and your templates match!

### CODE REVIEW

File: facts/secrets/openwrt-example.yaml [hunk 1]
```diff
@@ -21,7 +22,7 @@ wired:
   networks:
     # interface names cant have special characters
     - name: 'mgmt'
-      ifname: 'eth1.103 eth1.503'
+      ifname: 'eth1'
```
Comment (sarcasticadmin) [inline facts/secrets/openwrt-example.yaml]: I changed this in the example so when Im building the openwrt imgs I dont have to setup a vlan locally. I usually forget to do that and then spend a bit figuring out why I cant connect

**Comment by sarcasticadmin** [comment]
@owendelong ya i actually use a completely different `openwrt.yaml` for the show so thats not a problem.

Thats a good idea, Ill update the `gomplateignore` to account for all cases

**Comment by kylerisse** [comment]
Merging this. Will also capture new gomolate and test.sh into #130 . This looks good @sarcasticadmin thank you!

**Comment by sarcasticadmin** [comment]
@kylerisse thanks! Ill keep an eye out the additions to #130

**Pull request merged**